Map GuiSettingsVideo and related classes

maxpowa opened the pull request:
Also changed misspelled `SENSETIVITY` to `SENSITIVITY` in GameSettings & related places.

maxpowa committed:
Map GuiSettingsVideo and related classes

copygirl commented on mappings/net/minecraft/client/gui/menu/GuiSettingsVideo.mapping:
```diff
@@ -2,6 +2,8 @@ CLASS none/bgv net/minecraft/client/gui/menu/GuiSettingsVideo
 	FIELD a title Ljava/lang/String;
 	FIELD f parent Lnone/bgn;
 	FIELD g settings Lnone/bdt;
+	FIELD h widgetList Lnone/bex;
```
Don't GUIs already have a collection of widgets? Is this special in some way? If so the name should reflect that.

maxpowa replied:
This is a `WidgetListMulti` instance, it is the container for the 2 column button view.

copygirl commented on mappings/net/minecraft/client/settings/GameSettings.mapping:
```diff
@@ -13,9 +13,14 @@ CLASS none/bdt net/minecraft/client/settings/GameSettings
 		FIELD L REALMS_NOTIFICATIONS Lnone/bdt$a;
 		FIELD M AUTO_JUMP Lnone/bdt$a;
+		FIELD N isSlider Z
+		FIELD O isBoolean Z
```
Fields should not have a `is` prefix or similar. `slider` and `switch` ?

maxpowa replied:
I was considering `toggle` but it didn't feel quite right. I guess I ended up prefixing these because what they indicate is somewhat confusing if they are just `slider` and `switch`. Being able to doc this would be nice :)

copygirl replied:
Oh, `toggle` is the one I couldn't think of! I'd actually prefer that tbh.

copygirl commented on mappings/net/minecraft/client/settings/GameSettings.mapping:
```diff
@@ -40,7 +45,36 @@ CLASS none/bdt net/minecraft/client/settings/GameSettings
+			ARG 0 value
+		METHOD a getOption (I)Lnone/bdt$a;
+			ARG 0 id
+		METHOD b isBoolean ()Z
```
`isSwitch` ?

asiekierka replied:
Or isBoolean? Depends on the usage of it...

copygirl replied:
Well if this corresponds to the field of the same name ... the field can't be named `boolean`. And I think switch is the closest thing I can think of that represents an on/off state (other than checkbox but that's usually rendered in a specific way).

maxpowa replied:
I've changed it to `isToggle` but it's still very confusing without documentation, both of the values (`toggle` and `slider`) are just booleans. They indicate whether or not the option is a numeric slider value option or a boolean value option.

maxpowa committed:
Change isSlider/isBoolean to slider/toggle

Still very confusing without documentation, both of the values are just booleans. They indicate whether or not the option is a numeric slider value option or a boolean value option.
